azure-communication-ui/azure-communication-ui/src/main/java/com/azure/android/communication/ui/utilities/FeatureFlags.java

```diff
@@ -0,0 +1,181 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.android.communication.ui.utilities;
+
+import android.app.Application;
+import android.content.Context;
+import android.content.SharedPreferences;
+import com.azure.android.communication.ui.R;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+/* Feature Flag Management
+
+There is 2 parts to this system.
+1) Enum entries (fixed features). These are features in the Composite Library itself
+2) Pluggable entries (dynamic features). This allows the Application (e.g. Demo, Contoso) to add
+   additional features using the system.
+
+Initialization:
+  The feature flag system requires an application context to access String and Boolean resources
+  as well as the SharedPreferences to retain choices
+
+  `FeatureFlags.initialize(context)` anywhere early in your code e.g. (activity/application).onCreate
+
+# Usage:
+  ## Checking Flags:
+  ### Built-in
+  `FeatureFlags.FlagName.active = true/false`
+  (enable/disable) a feature or check it's status
+
+  ### Add-on
+  `yourFeatureFlag.active = true/false`
+  (enable/disable) a feature or check it's status
+
+  ## Registering a feature from outside UI Composite Library
+  `FeatureFlags.registerAdditionalFeature(yourFeatureFlag)`
+
+  ## Reading all FeatureFlags (Enum + Addon)
+  `FeatureFlags.features`
+  This is recommended over "values()" which will only access the Enum values.
+
+ */
+
+public enum FeatureFlags implements FeatureFlag {
```

Comment:
Azure review team suggested to use ExpandableStringEnum. I am not sure if we will get same feedback from azure review team.